experimental/sql/src/ibis_to_alg.py

```diff
@@ -0,0 +1,107 @@
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from dataclasses import dataclass
+from xdsl.ir import Operation, MLContext, Region, Block
+from typing import List, Type, Optional
+from xdsl.dialects.builtin import ArrayAttr, StringAttr, ModuleOp, IntegerAttr
+
+from xdsl.pattern_rewriter import RewritePattern, GreedyRewritePatternApplier, PatternRewriteWalker, PatternRewriter, op_type_rewrite_pattern
+
+import dialects.ibis_dialect as ibis
+import dialects.rel_alg as RelAlg
+
+# This file defines several `Rewriter`s that translate ibis dialect operations to
+# the corresponding constructs in the relational algebra dialect. The
+# `IbisRewriter` defines the parent class that contains all functions used by
+# several specific `Rewriter`s (like convert_datatype). All specific `Rewriter`s
+# inherit from the IbisRewriter.
+
+
+@dataclass
+class IbisRewriter(RewritePattern):
+
+  def convert_datatype(self, type_: ibis.DataType) -> RelAlg.DataType:
+    if isinstance(type_, ibis.String):
+      return RelAlg.String.get(type_.nullable)
+    if isinstance(type_, ibis.Int32):
+      return RelAlg.Int32()
+    raise Exception(
+        f"datatype conversion not yet implemented for {type(type_)}")
+
+
+@dataclass
+class SchemaElementRewriter(IbisRewriter):
```

Comment:
Some comment that this is a simple 1-1 rewrite.
In C++ we have something resembling `OneToOneRewriter<SourceTy, TargetTy>` in a few places.